test/utils/net.go

```diff
@@ -37,9 +46,17 @@ PORTS:
 		}
 
 		if returnErr != nil {
-			t.Fatalf("Time out: reached max %d retries. Error: %v", maxRetry, returnErr)
+			if t != nil {
+				t.Fatalf("Time out: reached max %d retries. Error: %v", maxRetry, returnErr)
+			} else {
+				log.Fatalf("Time out: reached max %d retries. Error: %v", maxRetry, returnErr)
+			}
 		} else {
-			t.Fatalf("Time out: reached max %d retries.", maxRetry)
+			if t != nil {
+				t.Fatalf("Time out: reached max %d retries.", maxRetry)
+			} else {
+				log.Fatalf("Time out: reached max %d retries.", maxRetry)
+			}
```

Comment:
You can use these methods: https://github.com/canonical/edgex-snap-testing/blob/06575b56e2d42c86edc3917e4a7ec63966df5719/test/utils/exec.go#L81-L101